openpype/modules/extra_addons/process_monitor/addon.py

```diff
@@ -0,0 +1,704 @@
+        timer_running = False
+        if self._timers_manager_module is not None:
+            timer_running = self._timers_manager_module.is_running
+
+        #TODO: useless variable, same as 'proc' (?)
+        terminated_process = None
+        #TODO: warning if not found?
+        if proc.pid in self._running_processes:
+            terminated_process = self._running_processes.pop(proc.pid)
+
+            process_parameters = [
+                #"name",
+                "cmdline"
+            ]
+            for process in psutil.process_iter(process_parameters):
+                pid = process.pid
+                if pid == 0:
+                    continue
+
+                #self.log.debug("name : {}".format(process.info["name"]))
+
+
+                ppid = process.ppid()
+                #self.log.debug("ppid : {}".format(ppid))
+                if ppid != proc.pid:
+                    continue
+
+                # Check if have command line of parent
+                # => To avoid children daemon processes spawned in background
+                #  to be considered as running processes,
+                #  such as 'ADPClientService' for Maya/3dsMax.
+                if not process.info["cmdline"]:
+                    continue
+                cmdline = [cmd.replace("\\", "/")
+                           for cmd in process.info["cmdline"]]
+
+                process_cmd = terminated_process["cmdline"]
+                #TODO: OK? (or should only check 'cmdline[0]'?
+                if process_cmd not in cmdline:
+                    continue
+
+                self.log.info((
+                    "ppid '{}' running => Replacing with child '{}'"
+                ).format(ppid, pid))
+
+                # Update current pid if was the terminated process
+                if self._current_pid == ppid:
+                    self._current_pid = pid
+
+                # Start new thread
+                process_thread = StoppableThread(process, self.on_terminate)
+                process_thread.start()
+                self.log.info((
+                    "Adding pid '{}' for '{}'"
+                ).format(pid, terminated_process["name"]))
+                self._running_processes[pid] = {
+                    "data": terminated_process["data"],
+                    "name": terminated_process["name"],
+                    "thread": process_thread,
+                    "cmdline": process_cmd
+                }
+
+                # No change for running timer
+                #TODO: Could be a problem if application not closed by user
+                # => Timer activated in background without user knowing
+                if self._current_pid != pid:
+                    self.timer_stopped()
+
+                    #if not self._timers_manager_module._idle_stopped:
+                    if timer_running:
+                        #TODO(2-REC): need timeout? (avoid if possible)
+                        import time
```

Comment:
import of built-in module inside methods?

Reply:
Yes, bad practice: it was to keep the ```import``` next to the command that was probably going to be removed. Was not going to be final.